qiskit_experiments/library/characterization/half_angle.py

```diff
@@ -0,0 +1,123 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Half angle calibration."""
+
+from typing import Optional
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.providers import Backend
+
+from qiskit_experiments.framework import BaseExperiment, Options
+from qiskit_experiments.library.calibration.analysis import FineHalfAngleAnalysis
+
+
+class HalfAngle(BaseExperiment):
+    r"""A calibration experiment class to perform half angle calibration.
+
+    # section: overview
+
+        This experiment runs circuits that repeat blocks of :code:`sx - sx - y`
+        circuits inserted in a Ramsey type experiment, i.e. the full gate sequence is thus
+        :code:`Ry(π/2) - [sx - sx - y] ^ n - sx` where :code:`n` is varied, see [1]. This
+        gate sequence is designed to amplify X-Y axis errors. Such errors can occur due to
+        phase errors.
+
+        .. parsed-literal::
+
+                    ┌─────────┐┌────┐┌────┐┌───┐   ┌────┐┌────┐┌───┐┌────┐ ░ ┌─┐
+               q_0: ┤ Ry(π/2) ├┤ sx ├┤ sx ├┤ y ├...┤ sx ├┤ sx ├┤ y ├┤ sx ├─░─┤M├
+                    └─────────┘└────┘└────┘└───┘   └────┘└────┘└───┘└────┘ ░ └╥┘
+            meas: 1/════════════════════════════...═══════════════════════════╩═
+                                                                              0
+
+    # section: reference
+        .. ref_arxiv:: 1 1504.06597
+    """
+
+    __analysis_class__ = FineHalfAngleAnalysis
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        r"""Default values for the fine amplitude experiment.
+
+        Experiment Options:
+            repetitions (List[int]): A list of the number of times that the gate
+                sequence :code:`[sx sx y]` is repeated.
+        """
+        options = super()._default_experiment_options()
+        options.repetitions = list(range(15))
+
+        return options
+
+    @classmethod
+    def _default_analysis_options(cls) -> Options:
+        r"""Default analysis options.
+
+        If the rotation error is very small the fit may chose a d_theta close to
+        :math:`\pm\pi`. To prevent this we impose bounds on d_theta.
+        """
+        options = super()._default_analysis_options()
+        options.normalization = True
+        options.angle_per_gate = np.pi
```

Comment:
Is it worth putting a comment to say that these options are not intended to be changed? They are options of the more flexible base class but the way the half angle experiment is written they are fixed.